stumpy/snippets.py

```diff
@@ -83,22 +94,25 @@ def _get_all_profiles(
             f"Please try `m <= len(T) // 2`."
         )
 
+    if s is not None:
+        s = min(int(s), m)
+    else:
+        percentage = np.clip(percentage, 0.0, 1.0)
+        s = min(math.ceil(percentage * m), m)
+
```

Comment:
I moved this if-else block as we need to have `s` before computng `T_subseq_isconstant`.